trustfall_core/src/ir/mod.rs

```diff
@@ -49,8 +54,12 @@ pub struct EdgeParameters(
     #[serde(default, skip_serializing_if = "BTreeMap::is_empty")] pub BTreeMap<Arc<str>, FieldValue>,
 );
 
+/// IR of components of a query, containing information about the vertex ID
+/// of the root of the query, as well as well as maps of all vertices, edges,
+/// folds, and outputs of the query.
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct IRQueryComponent {
+    /// The [Vid] of the root, or entry point, of the query.
```

Comment:
```suggestion
    /// The [Vid] of the root, or entry point, of the component.
```
Because it can be the entry point of a subcomponent, not the top-level one. If you take a peek at `IRFold` (the one that's part of the `folds` field), you'll find another `IRQueryComponent` in there.